src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,888 @@
+                    self.state._set_value(e, force_write=True)
+                except Exception as e:
+                    # any other type of exception.
+                    logging.error(e)
+                finally:
+                    # reset message buffer to receive the next.
+                    msg = ''
+
+        except Exception as e:
+            # another exception. End running.
+            logging.error(e)
+            
+        finally: 
+            # called if shutdown flag is set and loop exits
+            logging.debug("Shutting down listening thread...")
+
+    def _actOnMessage(self,decoded_msg):
+        '''
+        Act on a message received. 
+        decoded_msg: A message dictionary that was decoded with self._decode
+        '''
+        if isinstance(decoded_msg, ExplainedStatusMessage):
+            # Received a status message with an explanation string.
+            # Acknowledge it and log it.
+            self._sendMessage(EncodedStatusReplyMessage(decoded_msg.ecStatus))
+            err_status = decoded_msg.ecStatus
+
+            if err_status < 0:
+                self._measurinement_stopped.set()
+                raise HwError("Error %d: %s" % (err_status, ERRCODE[err_status]))
+
+        elif isinstance(decoded_msg, EncodedStatusMessage):
+            # Received a status message with an encoded status
+            # Acknowledge it and log it
+
+            # Check status messages
+            err_status = decoded_msg.ecStatus
+            self._sendMessage(EncodedStatusReplyMessage(decoded_msg.ecStatus))
+
+            if err_status == PQ_ERRCODE_NO_ERROR:
+                # the measurement completed successfully.
+                self.state._set_value(model.ST_RUNNING, force_write=True)
+                    
+            elif err_status == PQ_ERRCODE_MEASUREMENT_READY:
+                # Measurement is complete
+                if not self._measurinement_stopped.is_set():
+                    logging.info("Acquisition completed successfully.")
+                    self._measurinement_stopped.set()
+                    # update metadata and send data array to subscribers
+                    md = self._metadata.copy()
+                    md[model.MD_DWELL_TIME] = self.scanner.dwellTime.value
+                    md[model.MD_ACQ_DATE] = self._tStart
+                    da = model.DataArray([[0]], md)
+                    self.data.notify(da)
+
+            else:
+                # All other status types denote errors. 
+                self._measurinement_stopped.clear()
+                raise SPTError(err_status)
+
+        elif isinstance(decoded_msg, DataframeServerAckMessage):
+            # A measurement data frame was received from the server
+            # Do not respond to this type.
+            if self._measurinement_stopped.is_set():
+                # We should not receive this message if not measuring
+                raise HwError("Measurement data frame received while not measuring. ")
```

Comment:
Let's not make such a fuss. Let's just a log a warning message and skip the message.